controllers/util/util.go

```diff
@@ -122,3 +126,77 @@ func GetSystemCerts() ([]byte, error) {
 	}
 	return data, err
 }
+
+func GetServiceHostname(ctx context.Context, svcName, ns string, client client.Client) (string, error) {
+	serviceHostname := ""
+	isAvailable, service, err := IsServiceHostnameAvailable(ctx, svcName, ns, client)
+	if err != nil {
+		return "", err
+	}
+
+	if isAvailable {
+		// Loop over all Service ports, if a secured port is found
+		// set port and scheme to its secured ones and exit the loop
+		servicePort := ""
+		serviceScheme := "http"
+		for port := range service.Spec.Ports {
```

Comment:
what if something weird happened and the Service had 0 ports? The loop would never run, `servicePort` would be an empy string, and your URL would end in a colon.

There are a couple ways you could guard against a malformed URL. Maybe set a default port, or maybe be really pedantic and check service.Name, service.Namespace, and port to not be empty below?

Reply:
`spec.ports` is required in Service, so if something weird happened it should fail before hitting this line, which I believe it's the idea behind the isServiceHostnameAvailable function.